test_projects/documented/c_src/example/example.spec.exs

```diff
@@ -0,0 +1,36 @@
+module Example
+
+interface [NIF]
+
+@doc """
+Test begin_file_documented_function documentation
+"""
+spec begin_file_documented_function(num :: int) :: {:ok :: label, answer :: int}
+
+@doc """
+Test inside_file_documented_function documentation
+"""
+spec inside_file_documented_function(num :: int) :: {:ok :: label, answer :: int}
+
+@doc """
+Invalid test invalid_double_documented_function documentation
+"""
+@doc """
+Test invalid_double_documented_function documentation
+"""
+spec invalid_double_documented_function(num :: int) :: {:ok :: label, answer :: int}
+
+spec inside_file_undocumented_function(num :: int) :: {:ok :: label, answer :: int}
+
+@doc """
+Test after_undocumented_documented_function documentation
+"""
+spec after_undocumented_documented_function(num :: int) :: {:ok :: label, answer :: int}
+
+@doc false
+spec undocumented_false_function(num :: int) :: {:ok :: label, answer :: int}
+
+@doc """
+Test end_file_documented_function documentation
+"""
+spec end_file_documented_function(num :: int) :: {:ok :: label, answer :: int}
```

Comment:
I would rather add documentation to existing testing projects, instead of creating new project

Reply:
I wanted to have separate tests for the documentation functionality. 

I needed to test many separate aspects of documentation parsing and didn't want to bloat other test projects with the junky documentation.

Reply:
I think that it is not a good idea, to create new testing project for every new feature, that we add and implementing `@doc` is not enough separate from rest of the features in `unifex`, to give it a new testing project. Even if adding `@doc` to existing testing project made them little bit less pretty, it is better, than creating new testing project for documentation

Reply:
@mat-hek what are your thoughts?

Reply:
I'd add documentation to existing projects (at least to some functions) and add unit tests for these 'many separate aspects of documentation parsing'